[8.x] [ResponseOps][Rules] Create Rules APIs package (#214187)

umbopepato opened the pull request:
# Backport

This will backport the following commits from `main` to `8.x`:
 - [[ResponseOps][Rules] Create Rules APIs package (#214187)](https://github.com/elastic/kibana/pull/214187)

<!--- Backport version: 9.6.6 -->

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sorenlouv/backport)

<!--BACKPORT [{"author":{"name":"Umberto Pepato","email":"umbopepato@users.noreply.github.com"},"sourceCommit":{"committedDate":"2025-03-26T09:01:51Z","message":"[ResponseOps][Rules] Create Rules APIs package (#214187)\n\n## Summary\n\n- Creates a `@kbn/response-ops-rules-apis` package, following the\nproposed structure for ResponseOps Management Experiences package.\n- Moves relevant rules API fetchers and react-query hooks to the new\npackage.\n- Adds an internal variant of the `/api/alerting/rule_types` endpoint\n(`/internal/alerting/_rule_types`), that returns the same value as the\npublic one + the newly added internal [`solution`\nfield](https://github.com/elastic/kibana/issues/212017), that we don't\nwant to expose publicly.\n\n## Verification steps\n\n1. Create rules that fire alerts\n2. Verify the usages of the moved/changed hooks, with limited privileges\nas well (i.e. only `Rules Settings` but not `Stack Rules`):\n2.1. Stack management and Observability rules, rule details and alerts\npages\n2.2. Rules tab in the Connector editor flyout\n2.3. Alerts table row actions (••• icon)\n2.4. Tags filter in the rules list page\n3. Using the DevTools, compare the response of the public and internal\n`rule_types` endpoins:\n    ```\n    GET kbn:/api/alerting/rule_types\n    GET kbn:/internal/alerting/_rule_types\n    ```\nChecking that the `solution` field is present only in the internal one\n\n## References \n\nCloses #213059 \n\n### Checklist\n\n- [x] [Unit or functional\ntests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)\nwere updated or added to match the most common scenarios\n\n---------\n\nCo-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>","sha":"7aac590af4e245049f3865ea23ed88a179a80a28","branchLabelMapping":{"^v9.1.0$":"main","^v8.19.0$":"8.x","^v(\\d+).(\\d+).\\d+$":"$1.$2"}},"sourcePullRequest":{"labels":["release_note:skip","Team:ResponseOps","Team:obs-ux-management","backport:version","v9.1.0","v8.19.0"],"title":"[ResponseOps][Rules] Create Rules APIs package","number":214187,"url":"https://github.com/elastic/kibana/pull/214187","mergeCommit":{"message":"[ResponseOps][Rules] Create Rules APIs package (#214187)\n\n## Summary\n\n- Creates a `@kbn/response-ops-rules-apis` package, following the\nproposed structure for ResponseOps Management Experiences package.\n- Moves relevant rules API fetchers and react-query hooks to the new\npackage.\n- Adds an internal variant of the `/api/alerting/rule_types` endpoint\n(`/internal/alerting/_rule_types`), that returns the same value as the\npublic one + the newly added internal [`solution`\nfield](https://github.com/elastic/kibana/issues/212017), that we don't\nwant to expose publicly.\n\n## Verification steps\n\n1. Create rules that fire alerts\n2. Verify the usages of the moved/changed hooks, with limited privileges\nas well (i.e. only `Rules Settings` but not `Stack Rules`):\n2.1. Stack management and Observability rules, rule details and alerts\npages\n2.2. Rules tab in the Connector editor flyout\n2.3. Alerts table row actions (••• icon)\n2.4. Tags filter in the rules list page\n3. Using the DevTools, compare the response of the public and internal\n`rule_types` endpoins:\n    ```\n    GET kbn:/api/alerting/rule_types\n    GET kbn:/internal/alerting/_rule_types\n    ```\nChecking that the `solution` field is present only in the internal one\n\n## References \n\nCloses #213059 \n\n### Checklist\n\n- [x] [Unit or functional\ntests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)\nwere updated or added to match the most common scenarios\n\n---------\n\nCo-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>","sha":"7aac590af4e245049f3865ea23ed88a179a80a28"}},"sourceBranch":"main","suggestedTargetBranches":["8.x"],"targetPullRequestStates":[{"branch":"main","label":"v9.1.0","branchLabelMappingKey":"^v9.1.0$","isSourceBranch":true,"state":"MERGED","url":"https://github.com/elastic/kibana/pull/214187","number":214187,"mergeCommit":{"message":"[ResponseOps][Rules] Create Rules APIs package (#214187)\n\n## Summary\n\n- Creates a `@kbn/response-ops-rules-apis` package, following the\nproposed structure for ResponseOps Management Experiences package.\n- Moves relevant rules API fetchers and react-query hooks to the new\npackage.\n- Adds an internal variant of the `/api/alerting/rule_types` endpoint\n(`/internal/alerting/_rule_types`), that returns the same value as the\npublic one + the newly added internal [`solution`\nfield](https://github.com/elastic/kibana/issues/212017), that we don't\nwant to expose publicly.\n\n## Verification steps\n\n1. Create rules that fire alerts\n2. Verify the usages of the moved/changed hooks, with limited privileges\nas well (i.e. only `Rules Settings` but not `Stack Rules`):\n2.1. Stack management and Observability rules, rule details and alerts\npages\n2.2. Rules tab in the Connector editor flyout\n2.3. Alerts table row actions (••• icon)\n2.4. Tags filter in the rules list page\n3. Using the DevTools, compare the response of the public and internal\n`rule_types` endpoins:\n    ```\n    GET kbn:/api/alerting/rule_types\n    GET kbn:/internal/alerting/_rule_types\n    ```\nChecking that the `solution` field is present only in the internal one\n\n## References \n\nCloses #213059 \n\n### Checklist\n\n- [x] [Unit or functional\ntests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)\nwere updated or added to match the most common scenarios\n\n---------\n\nCo-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>","sha":"7aac590af4e245049f3865ea23ed88a179a80a28"}},{"branch":"8.x","label":"v8.19.0","branchLabelMappingKey":"^v8.19.0$","isSourceBranch":false,"state":"NOT_CREATED"}]}] BACKPORT-->

umbopepato committed:
[ResponseOps][Rules] Create Rules APIs package (#214187)

## Summary

- Creates a `@kbn/response-ops-rules-apis` package, following the
proposed structure for ResponseOps Management Experiences package.
- Moves relevant rules API fetchers and react-query hooks to the new
package.
- Adds an internal variant of the `/api/alerting/rule_types` endpoint
(`/internal/alerting/_rule_types`), that returns the same value as the
public one + the newly added internal [`solution`
field](https://github.com/elastic/kibana/issues/212017), that we don't
want to expose publicly.

## Verification steps

1. Create rules that fire alerts
2. Verify the usages of the moved/changed hooks, with limited privileges
as well (i.e. only `Rules Settings` but not `Stack Rules`):
2.1. Stack management and Observability rules, rule details and alerts
pages
2.2. Rules tab in the Connector editor flyout
2.3. Alerts table row actions (••• icon)
2.4. Tags filter in the rules list page
3. Using the DevTools, compare the response of the public and internal
`rule_types` endpoins:
    ```
    GET kbn:/api/alerting/rule_types
    GET kbn:/internal/alerting/_rule_types
    ```
Checking that the `solution` field is present only in the internal one

## References

Closes #213059

### Checklist

- [x] [Unit or functional
tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html)
were updated or added to match the most common scenarios

---------

Co-authored-by: kibanamachine <42973632+kibanamachine@users.noreply.github.com>
(cherry picked from commit 7aac590af4e245049f3865ea23ed88a179a80a28)

# Conflicts:
#	.github/CODEOWNERS
#	x-pack/platform/plugins/shared/alerting/common/routes/rule/apis/list_types/schemas/v1.ts
#	x-pack/test/alerting_api_integration/security_and_spaces/group1/tests/alerting/rule_types_internal.ts
#	x-pack/test/alerting_api_integration/security_and_spaces/group5/tests/alerting/index.ts
#	x-pack/test/alerting_api_integration/spaces_only/tests/alerting/group1/get_rule_tags.ts

elasticmachine commented:
Pinging @elastic/obs-ux-management-team (Team:obs-ux-management)

elasticmachine commented:
## :broken_heart: Build Failed
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/287241)
* Commit: f09e7da4381d3f9c291571d4545bdb963d749929

### Failed CI Steps
* [Check Types](https://buildkite.com/elastic/kibana-pull-request/builds/287241#0195d1f4-b8b4-462e-8785-f1ed46ec2943)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287241","number":287241,"commit":"f09e7da4381d3f9c291571d4545bdb963d749929"}],"number":287241}
buildkite-pr-comment-->

umbopepato committed:
Fix usage of old rule types api

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/lint_ts_projects --fix'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/287276)
* Commit: 4c40417995ddd0993ec9a7def0c4e4f96da65e6f

### Failed CI Steps
* [Build Kibana Distribution](https://buildkite.com/elastic/kibana-pull-request/builds/287276#0195d25b-954a-4809-b6fc-c1c5c385e6b0)

### History
* :broken_heart: [Build #287241](https://buildkite.com/elastic/kibana-pull-request/builds/287241) failed f09e7da4381d3f9c291571d4545bdb963d749929

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287276","number":287276,"commit":"4c40417995ddd0993ec9a7def0c4e4f96da65e6f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287241","number":287241,"commit":"f09e7da4381d3f9c291571d4545bdb963d749929"}],"number":287276}
buildkite-pr-comment-->

umbopepato committed:
Remove unnecessary plugin dependency

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/287280)
* Commit: 2bd47361a0ec0eb2d05c7eb8a0d70694e53124ab

### Failed CI Steps
* [Jest Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/287280#0195d28c-ebbb-483b-9d59-58ca7ce4a2e5)
* [Jest Tests #2](https://buildkite.com/elastic/kibana-pull-request/builds/287280#0195d2b2-1edd-44f7-9409-5e0fe6794d87)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/287280#0195d28c-ebbb-483b-9d59-58ca7ce4a2e5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/287280/jobs/0195d28c-ebbb-483b-9d59-58ca7ce4a2e5/artifacts/0195d2b2-0674-41eb-b43b-e2803e727ceb) Jest Tests #<span></span>2 / useGetRuleTypesPermissions should filter out rule types not present in `registeredRuleTypes`
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/287280#0195d2b2-1edd-44f7-9409-5e0fe6794d87) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/287280/jobs/0195d2b2-1edd-44f7-9409-5e0fe6794d87/artifacts/0195d2b9-8648-42a0-bdfe-050fd375786f) Jest Tests #<span></span>2 / useGetRuleTypesPermissions should filter out rule types not present in `registeredRuleTypes`
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/287280#0195d28c-ebbb-483b-9d59-58ca7ce4a2e5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/287280/jobs/0195d28c-ebbb-483b-9d59-58ca7ce4a2e5/artifacts/0195d2b2-0673-40b3-a564-6416233f247f) Jest Tests #<span></span>2 / useGetRuleTypesPermissions should not filter the rule types if `filteredRuleTypes` and `registeredRuleTypes` are not defined
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/287280#0195d2b2-1edd-44f7-9409-5e0fe6794d87) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/287280/jobs/0195d2b2-1edd-44f7-9409-5e0fe6794d87/artifacts/0195d2b9-8648-4d02-94d6-6f54e0760272) Jest Tests #<span></span>2 / useGetRuleTypesPermissions should not filter the rule types if `filteredRuleTypes` and `registeredRuleTypes` are not defined
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/287280#0195d28c-ebbb-483b-9d59-58ca7ce4a2e5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/287280/jobs/0195d28c-ebbb-483b-9d59-58ca7ce4a2e5/artifacts/0195d2b2-0674-4928-a621-6c4a030bf3a5) Jest Tests #<span></span>2 / useGetRuleTypesPermissions should return the correct authz flags for read-only rule types
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/287280#0195d2b2-1edd-44f7-9409-5e0fe6794d87) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/287280/jobs/0195d2b2-1edd-44f7-9409-5e0fe6794d87/artifacts/0195d2b9-8644-4c99-a2b1-fe0cacd620ce) Jest Tests #<span></span>2 / useGetRuleTypesPermissions should return the correct authz flags for read-only rule types
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/287280#0195d28c-ebbb-483b-9d59-58ca7ce4a2e5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/287280/jobs/0195d28c-ebbb-483b-9d59-58ca7ce4a2e5/artifacts/0195d2b2-0673-4adb-b6d4-c37a13eab09c) Jest Tests #<span></span>2 / useGetRuleTypesPermissions should return the correct authz flags for read+write rule types
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/287280#0195d2b2-1edd-44f7-9409-5e0fe6794d87) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/287280/jobs/0195d2b2-1edd-44f7-9409-5e0fe6794d87/artifacts/0195d2b9-8647-470e-b1fe-16e878d1d840) Jest Tests #<span></span>2 / useGetRuleTypesPermissions should return the correct authz flags for read+write rule types
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/287280#0195d28c-ebbb-483b-9d59-58ca7ce4a2e5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/287280/jobs/0195d28c-ebbb-483b-9d59-58ca7ce4a2e5/artifacts/0195d2b2-0673-4b2a-b207-e4855f2964e2) Jest Tests #<span></span>2 / useGetRuleTypesPermissions should return the correct authz flags when no rule types are accessible
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/287280#0195d2b2-1edd-44f7-9409-5e0fe6794d87) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/287280/jobs/0195d2b2-1edd-44f7-9409-5e0fe6794d87/artifacts/0195d2b9-8649-4ac7-abf3-43978f259603) Jest Tests #<span></span>2 / useGetRuleTypesPermissions should return the correct authz flags when no rule types are accessible

### History
* :broken_heart: [Build #287276](https://buildkite.com/elastic/kibana-pull-request/builds/287276) failed 4c40417995ddd0993ec9a7def0c4e4f96da65e6f
* :broken_heart: [Build #287241](https://buildkite.com/elastic/kibana-pull-request/builds/287241) failed f09e7da4381d3f9c291571d4545bdb963d749929

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287280","number":287280,"commit":"2bd47361a0ec0eb2d05c7eb8a0d70694e53124ab"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287276","number":287276,"commit":"4c40417995ddd0993ec9a7def0c4e4f96da65e6f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287241","number":287241,"commit":"f09e7da4381d3f9c291571d4545bdb963d749929"}],"number":287280}
buildkite-pr-comment-->

umbopepato committed:
Fix test

umbopepato committed:
Merge branch '8.x' into backport/8.x/pr-214187

delanni committed:
[fix] Fix lint errors introduced parallel to arming the rule (#216047)

## Summary
Quick fix to a lint error present in main after
https://github.com/elastic/kibana/pull/212508 - probably a case of merge
race condition

see:
https://buildkite.com/elastic/kibana-on-merge/builds/65208#0195d2a4-1f65-4ecc-9c45-25b789d8ca79

elasticmachine commented:
## :green_heart: Build Succeeded
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/287417)
* Commit: 0851761145d1519a778dfe95894916ce795f921f

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/d716361dfe56af2e62507fe740193b05f9c2c272) | [after](https://github.com/elastic/kibana/commit/0851761145d1519a778dfe95894916ce795f921f) | diff |
| --- | --- | --- | --- |
| `alerting` | 226 | 229 | +3 |
| `apm` | 1927 | 1930 | +3 |
| `cases` | 1013 | 1016 | +3 |
| `discover` | 1249 | 1252 | +3 |
| `embeddableAlertsTable` | 281 | 286 | +5 |
| `infra` | 1747 | 1750 | +3 |
| `ml` | 2435 | 2440 | +5 |
| `monitoring` | 734 | 737 | +3 |
| `observability` | 1253 | 1258 | +5 |
| `observabilityLogsExplorer` | 431 | 434 | +3 |
| `securitySolution` | 7136 | 7139 | +3 |
| `slo` | 1125 | 1128 | +3 |
| `synthetics` | 1240 | 1243 | +3 |
| `transform` | 717 | 720 | +3 |
| `triggersActionsUi` | 835 | 839 | +4 |
| `uptime` | 813 | 816 | +3 |
| total |  |  | +55 |

#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/d716361dfe56af2e62507fe740193b05f9c2c272) | [after](https://github.com/elastic/kibana/commit/0851761145d1519a778dfe95894916ce795f921f) | diff |
| --- | --- | --- | --- |
| `@kbn/alerting-types` | 248 | 249 | +1 |
| `triggersActionsUi` | 525 | 520 | -5 |
| total |  |  | -4 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d716361dfe56af2e62507fe740193b05f9c2c272) | [after](https://github.com/elastic/kibana/commit/0851761145d1519a778dfe95894916ce795f921f) | diff |
| --- | --- | --- | --- |
| `alerting` | 94.4KB | 94.7KB | +276.0B |
| `apm` | 2.6MB | 2.6MB | +374.0B |
| `cases` | 1.3MB | 1.3MB | +287.0B |
| `discover` | 1003.7KB | 1004.1KB | +379.0B |
| `embeddableAlertsTable` | 803.8KB | 804.1KB | +390.0B |
| `infra` | 1.6MB | 1.6MB | +376.0B |
| `ml` | 5.4MB | 5.4MB | +416.0B |
| `monitoring` | 664.9KB | 665.2KB | +376.0B |
| `observability` | 1.3MB | 1.3MB | +224.0B |
| `observabilityLogsExplorer` | 268.4KB | 268.8KB | +482.0B |
| `securitySolution` | 9.1MB | 9.1MB | +295.0B |
| `slo` | 918.3KB | 918.6KB | +376.0B |
| `synthetics` | 1.1MB | 1.1MB | +376.0B |
| `transform` | 632.4KB | 632.8KB | +376.0B |
| `triggersActionsUi` | 1.5MB | 1.5MB | +665.0B |
| `uptime` | 533.9KB | 534.2KB | +376.0B |
| total |  |  | +5.9KB |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/d716361dfe56af2e62507fe740193b05f9c2c272) | [after](https://github.com/elastic/kibana/commit/0851761145d1519a778dfe95894916ce795f921f) | diff |
| --- | --- | --- | --- |
| `alerting` | 52 | 51 | -1 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/d716361dfe56af2e62507fe740193b05f9c2c272) | [after](https://github.com/elastic/kibana/commit/0851761145d1519a778dfe95894916ce795f921f) | diff |
| --- | --- | --- | --- |
| `observability` | 93.9KB | 93.9KB | -1.0B |
| `observabilityLogsExplorer` | 15.1KB | 14.9KB | -146.0B |
| `triggersActionsUi` | 108.5KB | 107.0KB | -1.5KB |
| total |  |  | -1.7KB |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/d716361dfe56af2e62507fe740193b05f9c2c272) | [after](https://github.com/elastic/kibana/commit/0851761145d1519a778dfe95894916ce795f921f) | diff |
| --- | --- | --- | --- |
| `@kbn/alerting-types` | 252 | 253 | +1 |
| `triggersActionsUi` | 531 | 526 | -5 |
| total |  |  | -4 |

#### ESLint disabled in files
| id | [before](https://github.com/elastic/kibana/commit/d716361dfe56af2e62507fe740193b05f9c2c272) | [after](https://github.com/elastic/kibana/commit/0851761145d1519a778dfe95894916ce795f921f) | diff |
| --- | --- | --- | --- |
| `@kbn/response-ops-rules-apis` | - | 2 | +2 |

#### ESLint disabled line counts
| id | [before](https://github.com/elastic/kibana/commit/d716361dfe56af2e62507fe740193b05f9c2c272) | [after](https://github.com/elastic/kibana/commit/0851761145d1519a778dfe95894916ce795f921f) | diff |
| --- | --- | --- | --- |
| `@kbn/test-suites-xpack` | 754 | 756 | +2 |

#### Total ESLint disabled count
| id | [before](https://github.com/elastic/kibana/commit/d716361dfe56af2e62507fe740193b05f9c2c272) | [after](https://github.com/elastic/kibana/commit/0851761145d1519a778dfe95894916ce795f921f) | diff |
| --- | --- | --- | --- |
| `@kbn/response-ops-rules-apis` | - | 2 | +2 |
| `@kbn/test-suites-xpack` | 780 | 782 | +2 |
| total |  |  | +4 |


</details>


### History
* :broken_heart: [Build #287280](https://buildkite.com/elastic/kibana-pull-request/builds/287280) failed 2bd47361a0ec0eb2d05c7eb8a0d70694e53124ab
* :broken_heart: [Build #287276](https://buildkite.com/elastic/kibana-pull-request/builds/287276) failed 4c40417995ddd0993ec9a7def0c4e4f96da65e6f
* :broken_heart: [Build #287241](https://buildkite.com/elastic/kibana-pull-request/builds/287241) failed f09e7da4381d3f9c291571d4545bdb963d749929

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"passed","success":true,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287417","number":287417,"commit":"0851761145d1519a778dfe95894916ce795f921f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287280","number":287280,"commit":"2bd47361a0ec0eb2d05c7eb8a0d70694e53124ab"},{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287276","number":287276,"commit":"4c40417995ddd0993ec9a7def0c4e4f96da65e6f"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/287241","number":287241,"commit":"f09e7da4381d3f9c291571d4545bdb963d749929"}],"number":287417}
buildkite-pr-comment-->
